File: app/src/main/java/com/example/macarus0/walkiewalkie/data/WalkLocationDao.java
package com.example.macarus0.walkiewalkie.data;


import android.arch.lifecycle.LiveData;
import android.arch.persistence.room.Dao;
import android.arch.persistence.room.Insert;
import android.arch.persistence.room.Query;

import java.util.List;

@Dao
public interface WalkLocationDao {

    @Query("select * from walklocation where walkId = :walkId order by locationId ASC")
    List<WalkLocation> getWalkLocations(long walkId);

    @Query("select * from walklocation where walkId = :walkId order by locationId ASC")
    LiveData<List<WalkLocation>> getLiveWalkLocations(long walkId);

    @Insert
    void addWalkLocations(List<WalkLocation> walkLocations);

}

File: app/src/main/java/com/example/macarus0/walkiewalkie/data/WalkLocationDao.kt
package com.example.macarus0.walkiewalkie.data


import androidx.lifecycle.LiveData
import androidx.room.Dao
import androidx.room.Insert
import androidx.room.Query

@Dao
interface WalkLocationDao {

    @Query("select * from walklocation where walkId = :walkId order by locationId ASC")
    fun getLiveWalkLocations(walkId: Long): LiveData<List<WalkLocation>>

    @Insert
    fun addWalkLocations(walkLocations: List<WalkLocation>)

}
